src/index.js

```diff
@@ -70,11 +70,19 @@ function splitFolder(folder) {
  *
  * @param {string} parentFolderId Id of the parent directory
  * @param {string | null} childFolderPath
+ * @param {number} depth Current recursion depth
  * @returns {Promise<string>}
  */
-async function getUploadFolderId(parentFolderId, childFolderPath) {
+async function getUploadFolderId(parentFolderId, childFolderPath, depth = 0) {
+    const MAX_DEPTH = 15;
+
+    if (depth > MAX_DEPTH) {
+        throw new Error(`Maximum folder traversal depth of ${MAX_DEPTH} reached, potential cyclic structure detected.`);
```

Comment:
These is no reason to assume that there is a cyclical folder structure here, I would remove that from the log

Reply:
Why dont you think there is a reason.
My understanding is that this is a recursive call. 
`depth` is counting up the number of current level you have traversed.

Reply:
There is some reason, and it could be this, but it is not necessarily that. I might be a very nested directory structure. It is qualified with "potential", so if we think that is the most likely, the it may be an appropriate comment. I would lean away from that but can accept it if you think it is sensible.

Reply:
I focused on the work potential, I think I have seen this approach of a max depth commonly enough. I maybe naively accept it.

Another approach is that we could test
currentFolderId exists in a list of visiting folder paths.

But then I am wondering if this is needed at all.

The function takes a folderId and a path.
There is a way to put a "cyclic directory structure" in to a path?